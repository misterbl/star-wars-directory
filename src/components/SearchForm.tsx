import React, { PureComponent } from "react";
import { FormikProps } from "formik";
export interface ISearchForm {
  searchTerm: string;
}

class SearchForm extends PureComponent<FormikProps<ISearchForm>> {
  onsearchTermChange = (
    event: Pick<React.ChangeEvent<HTMLInputElement>, "target">
  ) => {
    this.props.setFieldValue("searchTerm", event.target.value);
  };

  render() {
    const { values, handleSubmit } = this.props;

    return (
      <form onSubmit={handleSubmit}>
        <label htmlFor="searchTerm" />
        <input
          autoComplete="off"
          placeholder="SEARCH STAR WARS"
          id="searchTerm"
          className="form-control mb-4"
          type="text"
          name="searchTerm"
          value={values.searchTerm}
          onChange={this.onsearchTermChange}
        />
        <button type="submit" className="search-button" />
      </form>
    );
  }
}
export default SearchForm;
