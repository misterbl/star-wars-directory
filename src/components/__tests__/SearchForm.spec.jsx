import * as React from "react";
import { shallow } from "enzyme";
import SearchForm from "../SearchForm";

describe("SearchForm", () => {
  const props = {
    setFieldValue: jest.fn(),
    values: {
      searchTerm: "searchTerm"
    },
    handleSubmit: jest.fn()
  };
  const wrapper = shallow(<SearchForm {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls setFieldValue on input change", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: "value" } });
    expect(props.setFieldValue).toHaveBeenCalledWith("searchTerm", "value");
  });
  it("calls handleSubmit on form submit", () => {
    const form = wrapper.find("form");
    form.simulate("submit");
    expect(props.handleSubmit).toHaveBeenCalled();
  });
});
