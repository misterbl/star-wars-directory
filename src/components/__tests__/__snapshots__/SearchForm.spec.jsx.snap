// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    handleSubmit={[MockFunction]}
    setFieldValue={[MockFunction]}
    values={
      Object {
        "searchTerm": "searchTerm",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="searchTerm"
        />,
        <input
          autoComplete="off"
          className="form-control mb-4"
          id="searchTerm"
          name="searchTerm"
          onChange={[Function]}
          placeholder="SEARCH STAR WARS"
          type="text"
          value="searchTerm"
        />,
        <button
          className="search-button"
          type="submit"
        />,
      ],
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "htmlFor": "searchTerm",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "className": "form-control mb-4",
          "id": "searchTerm",
          "name": "searchTerm",
          "onChange": [Function],
          "placeholder": "SEARCH STAR WARS",
          "type": "text",
          "value": "searchTerm",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-button",
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="searchTerm"
          />,
          <input
            autoComplete="off"
            className="form-control mb-4"
            id="searchTerm"
            name="searchTerm"
            onChange={[Function]}
            placeholder="SEARCH STAR WARS"
            type="text"
            value="searchTerm"
          />,
          <button
            className="search-button"
            type="submit"
          />,
        ],
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "htmlFor": "searchTerm",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "className": "form-control mb-4",
            "id": "searchTerm",
            "name": "searchTerm",
            "onChange": [Function],
            "placeholder": "SEARCH STAR WARS",
            "type": "text",
            "value": "searchTerm",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-button",
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
